### ISSUE TITLE
C# 7

### ISSUE THREAD

**Issue by jnm2** [nunit/nunit3-vs-adapter#451]
While working on https://github.com/nunit/nunit3-vs-adapter/issues/296, I'm running into cases where I want to use default literals or `?? throw`  or `is var` and `out var`. For example:

```cs
while (fullName.TryGetNextNamespaceSegment() is var segment && segment != null)
{
    // ...
}
```

Instead of:

```cs
for (;;)
{
    var segment = fullName.TryGetNextNamespaceSegment();
    if (segment == null) break;
    // ...
}
```

Since none of the projects have an explicit version set, and the solution already requires VS2017, and @oznetmaster doesn't transpile this codebase to C# 3, it seems like C# 7 is guaranteed to be available to anyone in the code. Does this mean we accept C# 7.0 code?

What about using `<LangVersion>latest</LangVersion>` or `7.1` or `7.2` so that I can use default literals or readonly structs? C# 7.1 requires VS 15.3 (or a package reference) and C# 7.2 requires VS 15.5 (or a package reference).

**Comment by OsirisTerje** [comment]
C# 7, yes.  Latest, fine by me.

**Comment by jnm2** [comment]
You know what, I did mean this just as a question but I'm happy to help if there's something you had in mind.

**Comment by OsirisTerje** [comment]
1. Adding <LangVersion>latest</LangVersion> to all projects.     As discussed earlier, there is no reason not to stay on the latest stable. If a problem arise from that, we just drop down to nearest downwards stable at that time.  

But we should convert all to packagereference format first.

**Comment by jnm2** [comment]
Sure. Is that something you would like me to do?

**Comment by jnm2** [comment]
@OsirisTerje Every csproj except the VSIX csproj is already using the new csproj SDK and PackageReference.
Last time I tried to convert a VSIX project to the new csproj SDK it didn't work. Is that something we should look into again?

[Oh hey, look!](https://github.com/nunit/nunit3-vs-adapter/pull/464/commits/7211a58f49bbc851384d70fb45fcb2cba596d04f) I forgot I already added <LangVersion> latest to all the projects with C# source code, except empty-assembly and mock-assembly. Think that's good enough?
